packages/runed/src/lib/utilities/useResizeObserver/useResizeObserver.svelte.ts

```diff
@@ -0,0 +1,93 @@
+import { IsSupported } from "../IsSupported/IsSupported.svelte.js";
+import { extract } from "../extract/extract.js";
+import { watch } from "../watch/watch.svelte.js";
+import { safelyCleanup } from "$lib/internal/utils/effect.svelte.js";
+import type { MaybeGetter } from "$lib/internal/types.js";
+
+export interface ResizeObserverSize {
+	readonly inlineSize: number;
+	readonly blockSize: number;
+}
+
+export interface ResizeObserverEntry {
+	readonly target: Element;
+	readonly contentRect: DOMRectReadOnly;
+	readonly borderBoxSize?: ReadonlyArray<ResizeObserverSize>;
+	readonly contentBoxSize?: ReadonlyArray<ResizeObserverSize>;
+	readonly devicePixelContentBoxSize?: ReadonlyArray<ResizeObserverSize>;
+}
+
+export type ResizeObserverCallback = (
+	entries: ReadonlyArray<ResizeObserverEntry>,
+	observer: ResizeObserver
+) => void;
+
+export interface UseResizeObserverOptions {
+	/**
+	 * Sets which box model the observer will observe changes to. Possible values
+	 * are `content-box` (the default), `border-box` and `device-pixel-content-box`.
+	 *
+	 * @default 'content-box'
+	 */
+	box?: ResizeObserverBoxOptions;
+}
+
+declare class ResizeObserver {
+	constructor(callback: ResizeObserverCallback);
+	disconnect(): void;
+	observe(target: Element, options?: UseResizeObserverOptions): void;
+	unobserve(target: Element): void;
+}
+
+/**
+ * Reports changes to the dimensions of an Element's content or the border-box
+ *
+ * @see https://runed.dev/docs/utilities/useResizeObserver
+ */
+export function useResizeObserver(
+	target: MaybeGetter<HTMLElement | HTMLElement[] | null | undefined>,
+	callback: ResizeObserverCallback,
+	options: UseResizeObserverOptions = {}
+) {
+	let observer: ResizeObserver | undefined;
+	const isSupported = new IsSupported(() => window && "ResizeObserver" in window);
```

Comment:
ResizeObserver has been supported by all major browsers for 3 years now. Do we need all the guard rails?